### DISCUSSION TITLE
Directus SDK File Upload - "Error: Network Error"

### DISCUSSION THREAD

**Discussion by ldrach** [directus/directus#16509] [Archive]
I just wanted to see if anyone else has come across this error when uploading a file using the Directus SDK.

`Error: Network Error`

I have a nuxt project that has a form where the user can upload documents. Most of the time, everything works as expected. I am using Sentry for error monitoring and I noticed that every now and then, there is an error (Error: Network Error) when uploading files and I can't seem to reproduce it. My function for file uploads is below:

```
async handleDocuments($event) {
        const formData = new FormData()
        this.errorMessage = ''
        formData.append(
          'title',
          this.fileTitle
        )
        formData.append('folder', this.formStore.applicant_folder)
        formData.append('file', $event)

        await this.$directus.files
          .createOne(formData)
          .then((resp) => {
            this.$emit('input', resp.id)
          })
          .catch((error) => {
            this.errorMessage = error
            this.$sentry.captureException(error)
          })
    }
```

There are a few spots in the form where a user can upload documents. From what I can tell in sentry, if a user gets this error, they continue to get an error when uploading a file until they exit the form, refresh, and then start the form again.

The strangest part about this is that the files are actually uploaded properly most of the time even though the response is an error. The error prevents the user from getting the uploaded files ID which is required for submitting the form so they are forced to restart the form.

Any insight on this issue would be greatly appreciated!

**Comment by rijkvanzanten** [comment]
A generic network error means that the error wasn't thrown by directus itself, so it could be anything else in between directus and the user. Even something like the network connection getting interrupted when the user went through a tunnel on a train would cause one of those generic network errors 🤔

**Reply by ldrach** [reply]
Thanks for the information! That definitely seems to line up with what I find out so far but I still wasn't 100% positive. That should narrow down the search at least!

**Reply by ldrach** [reply]
The part that made me unsure about if directus was throwing the error was the log in Sentry that I probably should've included. This is supposed to be where the error was thrown

```
./node_modules/@directus/sdk/dist/sdk.bundler.js in TransportError at line 499:13

        var _a, _b;
        if ((_a = response === null || response === void 0 ? void 0 : response.errors) === null || _a === void 0 ? void 0 : _a.length) {
            super((_b = response === null || response === void 0 ? void 0 : response.errors[0]) === null || _b === void 0 ? void 0 : _b.message);
        }
        else {
            **super((parent === null || parent === void 0 ? void 0 : parent.message) || 'Unknown transport error');** <-- Line Throwing Error
        }
        this.parent = parent;
        this.response = response;
        this.errors = (response === null || response === void 0 ? void 0 : response.errors) || [];
        if (!Object.values(response || {}).some((value) => value !== undefined)) {
```

**Comment by AxeemHaider** [comment]
I'm getting the same error "Network Error" this is defiantly from directus when you try to upload file from react-native.

```js
const formaData = new FormData();
formData.append("title", "My File");
formData.append("file", {
  name: file.name,
  type: file.type,
  uri: file.uri
});

directus.files.createOne(formData); // Network Error
```

except this everything is working fine no **Network Error** or something this only happen when I try to run this peace of code
